java-tests/validation-gen/src/test/kotlin/io/spine/test/tools/validate/setonce/SetOnceAssertions.kt

```diff
@@ -24,13 +24,27 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+package io.spine.test.tools.validate.setonce
+
+import io.spine.validate.ValidationException
+import org.junit.jupiter.api.Assertions.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+
 /**
- * Contains tests for `(set_once)` field option.
+ * Assertions for `(set_once)` tests.
  */
-@CheckReturnValue
-@ParametersAreNonnullByDefault
-package io.spine.test.tools.validate.setonce;
+internal object SetOnceAssertions {
 
-import com.google.errorprone.annotations.CheckReturnValue;
+    /**
+     * Asserts that the given [runnable] throws [ValidationException].
+     */
+    fun assertValidationFails(runnable: () -> Unit) {
+        assertThrows<ValidationException>(runnable)
+    }
 
-import javax.annotation.ParametersAreNonnullByDefault;
+    /**
+     * Asserts that the given [runnable] doesn't throw anything.
+     */
+    fun assertValidationPasses(runnable: () -> Unit) =
+        assertDoesNotThrow(runnable)
```

Comment:
It is also useful to return a result of the executing the `runnable`. It helps when you need to verify the result of building a message. We also have tests that  run `.validate()` in combination with just building a message.

Again, please have a look around the code of the project.

Reply:
I mean the code like this:

```kotlin
val msg = assertDoesNotThrow {
    protoSet {
        element.add(toAny("42"))
        element.add(toAny(42))
    }
}
msg.validate().shouldBeEmpty()
```

Reply:
If we introduce something like `assertValid()` it may be implemented as the code above. But, again, we need to return the produced message so that the tests that use the utility can do something useful with it without the "gymnastics" with `var` declarations.

Reply:
I've introduced them within an object as `(set_once)`-specific intentionally.

1. Assertions within `IsValid` are not applicable as they rely on validation being done in `builder()` method. The test subject here throws right from a builder setter.
2. Within `(set_once)` tests, we don't need the results, so I don't return them, allowing the test cases be one-lines.

I didn't look at them as general-purpose. I can move them back to `Spec*` files (two files need them), and make `private`. Or we can discuss vocally.